src/runner/browser-env/vite/utils.ts

```diff
@@ -27,3 +39,39 @@ export const getNodeModulePath = async ({
 export const prepareError = (error: Error): Error => {
     return JSON.parse(JSON.stringify(error, Object.getOwnPropertyNames(error)));
 };
+
+export const getTestInfoFromViteRequest = (req: Connect.IncomingMessage): TestInfoFromViteRequest | null => {
+    if (!req.url?.endsWith("index.html") || !req.originalUrl) {
+        return null;
+    }
+
+    const parsedUrl = url.parse(req.originalUrl);
+    const [routeName, runUuid] = _.compact(parsedUrl.pathname?.split("/"));
+
+    if (routeName !== VITE_RUN_UUID_ROUTE || !runUuid) {
+        throw new Error(`Pathname must be in "/${VITE_RUN_UUID_ROUTE}/:uuid" format, but got: ${req.originalUrl}`);
+    }
+
+    const env = WORKER_ENV_BY_RUN_UUID.get(runUuid);
+    if (!env) {
+        throw new Error(
+            `Worker environment is not found by "${runUuid}". ` +
+                "This is possible if:\n" +
+                '  - "runUuid" is not generated by Testplane\n' +
+                "  - the test has already been completed\n" +
+                "  - worker was disconnected",
+        );
+    }
+
+    return { routeName, runUuid, env };
+};
+
+export const getPathWithoutExtName = (fsPath: string): string => {
+    const extname = path.extname(fsPath);
+
+    if (!extname) {
+        return fsPath;
+    }
+
+    return fsPath.slice(0, extname.length * -1);
```

Comment:
Why `extname.length * -1` and not `-extname.length`? `extname.length * -1` looks weird